### ISSUE TITLE
RGBW2 - switch sensor

### ISSUE THREAD

**Issue by hk-ret** [StyraHem/ShellyForHASS#80]
I'm not sure if this is the same as [issue 39](https://github.com/StyraHem/ShellyForHASS/issues/39).
I've added `switch` to the list of sensors, however for RGBW2 there is no switch sensor detected. For Shelly 1 and 2.5 this works fine.

**Comment by chemelli74** [comment]
@hk-ret , can you please retest with 0.1.6b6 ? Please update firmware to latest v1.5.7 as well.

Thank you,

Simone

**Comment by hk-ret** [comment]
Dear Simone
Unfortunately the problem still exists.
I have retested with shelly `0.1.6.b8` (pyShelly `0.1.17`). I couldn't find 0.1.6b6.
The Shelly RGBW2 runs firmware version `20191216-090536/v1.5.7@c30657ba`.
The shelly configuration looks like this (1 RGBW2 and 2 Shelly-1):
~~~
shelly:
  discovery: false
  sensors: [switch]
  version: true
  devices:
    - id: 5A381C
      name: RGBW2
      entity_id: rgbw
    - id: 500134
      name: Switch
      entity_id: switch
    - id: 944EE4
      name: Light
      entity_id: light
      light_switch: true
~~~
And these are all the shelly entities created:
![Shellies](https://user-images.githubusercontent.com/55021191/73124142-22d22e00-3f98-11ea-8821-fc7ac73ba09d.PNG)
The `binary_sensor`s are only created for the Shelly-1's. The same result is returned if the `sensors: [switch]` is added to each device separately.

Let me know if you need further information.

Cheers, Heinz

**Comment by hakana** [comment]
Yes, i got same result, will be fixed in next release, beta 9

**Comment by hakana** [comment]
Fixed in beta 9 just released

**Comment by hk-ret** [comment]
I confirm that the issue is fixed in 0.1.6.b9 

Thanks a lot.

**Comment by hakana** [comment]
Perfect!
